training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple-sharded-data-parallel.ipynb

```diff
@@ -0,0 +1,3380 @@
+    "\n",
+    "Please see the instructions from [Distributed Training of Mask-RCNN in Amazon SageMaker Using FSx](https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/distributed_tensorflow_mask_rcnn/mask-rcnn-scriptmode-fsx.ipynb) to create an FSx Lustre file system and import the dataset from the S3 bucket to your FSx file system. Note that the FSx file system must be created in a private subnet with internet gateway to ensure that training job has access to the internet. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Instructions obtained from:\n",
+    "# https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/distributed_tensorflow_mask_rcnn/mask-rcnn-scriptmode-fsx.ipynb\n",
+    "\n",
+    "if use_fsx:\n",
+    "    from sagemaker.inputs import FileSystemInput\n",
+    "\n",
+    "    # Specify FSx Lustre file system id.\n",
+    "    file_system_id = \"<your-file-system-id>\"\n",
+    "\n",
+    "    # Specify the SG and subnet used by the FSX, these are passed to SM Estimator so jobs use this as well\n",
+    "    fsx_security_group_id = \"<your-security-group-id>\"\n",
+    "    fsx_subnet = \"<your-subnet>\"\n",
+    "\n",
+    "    # Specify directory path for input data on the file system.\n",
+    "    # You need to provide normalized and absolute path below.\n",
+    "    # Your mount name can be provided by you when creating fsx, or generated automatically.\n",
+    "    # You can find this mount_name on the FSX page in console.\n",
+    "    # Example of fsx generated mount_name: \"3x5lhbmv\"\n",
+    "    base_path = \"<your-mount-name>\"\n",
+    "\n",
+    "    # Specify your file system type.\n",
+    "    file_system_type = \"FSxLustre\"\n",
+    "\n",
+    "    train = FileSystemInput(\n",
+    "        file_system_id=file_system_id,\n",
+    "        file_system_type=file_system_type,\n",
+    "        directory_path=base_path,\n",
+    "        file_system_access_mode=\"rw\",\n",
+    "    )\n",
+    "\n",
+    "    data_channels = {\"train\": train, \"test\": train}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Set Up Hyperparameters, Metric Definitions, and MPI Options\n",
+    "The following `hyperparameters` dictionary is to pass arguments to the training script (`train_gpt_simple.py`) and set the model parallel configuration when creating the training job.\n",
+    "\n",
+    "You can also add custom mpi flags. By default, we have `--mca btl_vader_single_copy_mechanism none` to remove unnecessary logs.\n",
+    "\n",
+    "Next, we add a base metric definitions to enable the metric upload in SageMaker. You can add any further metric definitions."
```

Comment:
Add a sentence to point out the new parameter is added as a hyperparameter. This should be the highlight of this notebook.